mautic/mautic
# Add shift+? shortcut to display modal of all keyboard shortcuts

Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | N
| New feature?  | Y
| BC breaks?    | N
| Deprecations? | N
| Fixed issues  |  N/A

## Description

On the mautic slack channel, we had a discussion regarding keyboard shortcuts and the fact that they aren't very well known. So I put this PR together to give users an overview of what shortcuts are available to them on a per/page basis. Whatever page you are on, you can see your available shortcuts by pressing `shift+?`

## Steps to reproduce the bug (if applicable)

Load Mautic. Press `shift+?`. Nothing happens and you have no idea what's going on.

## Steps to test this PR

Load the PR, then press `shift+?`, and you'll get a nice modal that shows all shortcuts for the page. Load the Contacts page, press the same key combo, and you'll see that there are contextual shortcuts added just for the contact page. Reload the dashboard, and the contextual shortcuts are gone.

Also, when testing, ensure that the modal still works for all other purposes.

![screen shot 2016-05-10 at 2 21 44 pm](https://cloud.githubusercontent.com/assets/718028/15158303/49274c5a-16bd-11e6-8292-82ac9a02d6af.png)
